tests/test_jax_utils.py

```diff
@@ -133,3 +139,57 @@ def d(conf, pairs, box):
 
     assert neighbor_distances.shape == (n_confs, n_neighbor_pairs)
     assert np.sum(neighbor_distances < cutoff) == np.sum(full_distances < cutoff)
+
+
+def test_pairwise_distances_assertions():
+    with pytest.raises(AssertionError):
+        pairwise_distances(np.empty((4, 3)), np.empty((1, 1)))  # inconsistent box shape
+    _ = pairwise_distances(np.empty((4, 3)), np.empty((3, 3)))  # ok
+
+    with pytest.raises(AssertionError):
+        pairwise_distances(np.empty((4, 3)), np.empty((3, 3)), np.empty((3,)))  # inconsistent w_coords shape
+    _ = pairwise_distances(np.empty((4, 3)), np.empty((3, 3)), np.empty((4,)))  # ok
+
+
+finite_floats = partial(st.floats, allow_nan=False, allow_infinity=False, allow_subnormal=False)
+coordinates = arrays(np.float64, array_shapes(min_dims=2, max_dims=2), elements=finite_floats(-1e6, 1e6))
+
+
+@given(coordinates)
+@seed(2022)
+def test_pairwise_distances(x):
+    n, _ = x.shape
+    dij = pairwise_distances(x)
+    assert dij.shape == (n, n)
+    assert (dij >= 0.0).all()
+
+
+@st.composite
+def coords_box_w_triples(draw):
+    x = draw(coordinates)
+    n, d = x.shape
+    box_diag = draw(arrays(np.float64, d, elements=finite_floats(1e-6, 1e6)))
+    w = draw(arrays(np.float64, n, elements=finite_floats(-1e6, 1e6)))
+    return x, box_diag, w
+
+
+@given(coords_box_w_triples())
+@seed(2022)
+def test_pairwise_distances_periodic(coords_box_w):
+    x, box_diag, _ = coords_box_w
+    n, _ = x.shape
+    dij = pairwise_distances(x, np.diagflat(box_diag))
```

Comment:
Not sure I understand hypothesis well enough, but is this always creating a box small enough to impose PBCs?

Reply:
The coordinates and box are uncorrelated, so this should test some examples where we expect wrapping and some where we don't.

This got me thinking that it might be good practice to add explicit examples of cases that we want to ensure are tested. E.g. I think there's some (probably vanishingly small) probability that the arbitrary examples don't include a case with a small enough box to test PBCs. I added an explicit example in 2a657ab.